subprojects/docs/src/docs/userguide/dependencyManagementTerminology.adoc

```diff
@@ -0,0 +1,44 @@
+[[dependency_management_terminology]]
+== Dependency Management Terminology
+
+Dependency management comes with a wealth of terminology. Here you can find the most commonly-used terms including references to the user guide to learn about their practical application.
+
+[[sub:terminology_module]]
+Module::
+
+A piece of software that evolves over time e.g. link:https://github.com/google/guava[Google Guava]. Every module has a name. Each release of a module is optimally represented by a <<sub:terminology_version,version>>. For convenient consumption, modules can be hosted in a <<sub:terminology_repository,repository>>.
+
+[[sub:terminology_version]]
+Version::
+
+A <<sub:terminology_module,module>> indicates its revision with the help of a version. Each module version contains a distinct set of changes. In practice there's no limitation to the scheme of the version. Timestamps, numbers, special suffixes like GA are all allowed identifiers. The most widely-used versioning strategy is link:https://semver.org/[semantic versioning]. Semantic versioning is represented by the scheme _major_._minor_._patch_ e.g. `1.4.15`. Each part conveys a special meaning about the backwards-compatibility of the release.
+
+[[sub:terminology_dependency]]
+Dependency::
+
+A dependency is defined as a versioned or unversioned module. Dependencies may provide <<sub:terminology_dependency_metadata,metadata>> to indicate that they depend on other dependencies required for it work properly, so-called <<sub:terminology_transitive_dependency,transitive dependencies>>. For more information, see <<declaring_dependencies>>. Gradle's dependency resolution process can apply constraints declared in the build script to select a specific version.
+
+[[sub:terminology_transitive_dependency]]
+Transitive dependency::
+
+A transitive dependency is a dependency required by another dependency to work properly. Dependencies hosted on a <<sec:terminology_repository,repository>> can provide <<sub:terminology_dependency_metadata,metadata>> declaring transitive dependencies. By default, Gradle resolves transitive dependencies if defined by metadata. The behavior is highly customizable. For more information, see <<managing_transitive_dependencies>>.
+
+[[sub:terminology_dependency_metadata]]
+Dependency metadata::
+
+Dependencies can provide metadata. Metadata is the data that describes the dependency in more detail e.g. the coordinates for locating it in a repository, information about the project or required <<sub:terminology_transitive_dependency,transitive dependencies>>. In Maven the metadata file is called `pom.xml`, in Ivy it is called `ivy.xml`.
```

Comment:
"In Maven the metadata file is called `pom.xml`, in Ivy it is called `ivy.xml`." ->

Should be `.pom` instead of `pom.xml`.
And lets mention our metadata here as well. It's already possible to use it if you opt-in or use native:

> Maven repositories provide metadata through `.pom` files, Ivy repositories through `ivy.xml` files. Modules published with Gradle can provide additional metadata through `.module` files in both Maven and Ivy repositories.

Reply:
> Should be .pom instead of pom.xml.

done

> And lets mention our metadata here as well. It's already possible to use it if you opt-in or use native:

It may be better for you to make the change in https://github.com/gradle/gradle/pull/4089. It will allow you to also look for other places where this term is needed. I can merge this PR soon.

Reply:
Dependencies don't provide metadata. Releases of modules provide metadata.
That is, unless we are going to make "dependency" a synonym for "module version". I think that's wrong, since it doesn't seem right to refer to an arbitrary release of a module (e.g. `com.google:guava:18.0`) as a _dependency_. You don't publish a _dependency_. A _dependency_ is always referred to in the context of a relationship between 2 things.